.github/workflows/lint.yml

```diff
@@ -0,0 +1,168 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      clang: ${{ steps.filter.outputs.clang }}
+      md: ${{ steps.filter.outputs.md }}
+      python: ${{ steps.filter.outputs.python }}
+    steps:
+      - name: Detect changed files
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            clang:
+              - '**/*.c'
+              - '**/*.h'
+              - '**/meson.build'
+              - 'meson_options.txt'
+            md:
+              - '**/*.md'
+            python:
+              - '**/*.py'
+
+
+
+  common:
+    name: Common
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/centos/centos:stream9
+
+    steps:
+      - name: Enable additional repositories
+        run: |
+          dnf install -y dnf-plugin-config-manager
+
+          # EPEL is required for codespell
+          dnf install -y epel-release
+
+      - name: Install dependencies
+        run: |
+          dnf install \
+                  codespell \
+                  make \
+              -y
+
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Check SPDX headers
+        run: |
+          ./build-scripts/spdx-header-chech.sh
+
+      - name: Running codespell checker
+        run: |
+          make check-codespell
+
+
+
+  clang:
+    needs: changes
+    if: ${{ needs.changes.outputs.clang == 'true' }}
+    name: C language
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/centos/centos:stream9
+
+    steps:
+      - name: Enable additional repositories
+        run: |
+          dnf install -y dnf-plugin-config-manager
+
+          # CRB is required for meson
+          dnf config-manager -y --set-enabled crb
+
+      - name: Install dependencies
+        run: |
+          dnf install \
+                  clang-tools-extra \
+                  bzip2 \
+                  gcc \
+                  meson \
+                  selinux-policy-devel \
+                  systemd-devel \
+                  valgrind \
+              -y
+
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Building hirte
+        run: |
+          make build
+
+      - name: Running source code format check
+        run: |
+          make check-fmt
+
+      - name: Running static analysis
+        run: |
+          make lint-c
+
+      - name: Running unit tests
+        run: |
+          make test-with-valgrind
+
+      - name: Upload unit test logs
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-logs
+          path: ./builddir/meson-logs/testlog-valgrind.txt
```

Comment:
The unit tests are kind of hidden in the clang lint step (as previously in the lint step). I think it is fine when we run the unit tests always if some .c/.h file changes, but I am wondering it would be better to move them to their own step?!

Reply:
Extracted unit tests into standalone workflow